pid_ctrl/src/pid_ctrl.c

```diff
@@ -138,3 +138,13 @@ esp_err_t pid_update_parameters(pid_ctrl_block_handle_t pid, const pid_ctrl_para
     }
     return ESP_OK;
 }
+
+esp_err_t pid_reset_pre_error(pid_ctrl_block_handle_t pid)
+{
+    ESP_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, TAG, "invalid argument");
+    pid->integral_err = 0;
+    pid->previous_err1 = 0;
+    pid->previous_err2 = 0;
+    pid->last_output = 0;
+    return ESP_OK;
+}
```

Comment:
If we need a reset API, I would suggest to make it more generic. In some cases we might only need to clear the integral error to do a fast response, and in another case we might want to reset the last output to eliminate the influence from the last output. Even in some unconventional apps, user wants to tamper with these data.

Therefore, I propose to have a generic update API like this:
```c
// Define a intermediate_data structure in the header
typedef struct {
    float integral_err;
    float previous_err1;
    float previous_err2;
    float last_output;
} pid_ctrl_intermediate_data_t;

// Realize it in the source file
esp_err_t pid_update_Intermediate data(pid_ctrl_block_handle_t pid, const pid_ctrl_intermediate_data_t *Intermediate_data) {
    ESP_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, TAG, "invalid argument");
    pid->integral_err = Intermediate_data->integral_err;
    pid->previous_err1 = Intermediate_data->previous_err1;
    pid->previous_err2 = Intermediate_data->previous_err2;
    pid->last_output = Intermediate_data->last_output;
}
```
It's open for discussion~

And don't forget to have a new line in the end of this file.

Reply:
thank you for the review,  I think generic update API is a good idea